From f4a3a70f69e1dbccb6578f39ef47835098a04624 Mon Sep 17 00:00:00 2001
From: Nick Bowler <nbowler@draconx.ca>
Date: Tue, 17 Aug 2021 18:26:42 -0700
Subject: automake: fatal error on second AM_INIT_AUTOMAKE.

This change addresses https://bugs.gnu.org/50046.
Patch posted:
https://lists.gnu.org/archive/html/automake-patches/2021-08/msg00000.html

* m4/init.m4: Make attempts to expand AM_INIT_AUTOMAKE more than
once a fatal error at m4 time.
---
 m4/init.m4 | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/m4/init.m4 b/m4/init.m4
index c5807d211..b4686b140 100644
--- a/m4/init.m4
+++ b/m4/init.m4
@@ -28,6 +28,10 @@ m4_defn([AC_PROG_CC])
 # release and drop the old call support.
 AC_DEFUN([AM_INIT_AUTOMAKE],
 [AC_PREREQ([2.65])dnl
+m4_ifdef([_$0_ALREADY_INIT],
+  [m4_fatal([$0 expanded multiple times
+]m4_defn([_$0_ALREADY_INIT]))],
+  [m4_define([_$0_ALREADY_INIT], m4_expansion_stack)])dnl
 dnl Autoconf wants to disallow AM_ names.  We explicitly allow
 dnl the ones we care about.
 m4_pattern_allow([^AM_[A-Z]+FLAGS$])dnl
-- 
cgit v1.2.1

